curvefs/src/client/inode_wrapper.cpp

```diff
@@ -271,6 +274,20 @@ CURVEFS_ERROR InodeWrapper::UnLinkLocked() {
         inode_.set_ctime_ns(now.tv_nsec);
         inode_.set_mtime(now.tv_sec);
         inode_.set_mtime_ns(now.tv_nsec);
+        // newlink == 0 will be deleted at metasever
+        // dir will not update parent
+        // parent = 0 is useless
+        if (newnlink != 0 && inode_.type() != FsFileType::TYPE_DIRECTORY
+            && parent != 0) {
+            auto parents = inode_.mutable_parent();
+            for (auto iter = parents->begin(); iter != parents->end(); iter++) {
+                if (*iter == parent) {
+                    parents->erase(iter);
+                    break;
+                }
+            }
+        }
+
```

Comment:
why UnLinkLocked handle parent , but DecreaseNLink does not handle parent?

Reply:
> why UnLinkLocked handle parent , but DecreaseNLink does not handle parent?

The operation 'UnLinkLocked' is used for current Inode if nlink is decreased to 0 will be deleted. But 'DecreaseNLink' is only used to decrease parentInode nlink have no inode delete.